src/icon-api.js

```diff
@@ -94,6 +96,14 @@ export class IconApi {
     return blob
   }
 
+  /**
+   * @param {string} docId
+   */
+  async delete(docId) {
+    const { versionId } = await this.#dataType.getByDocId(docId)
+    return await this.#dataType.delete(versionId)
```

Comment:
await not needed in this case

```suggestion
    return this.#dataType.delete(versionId)
```